pkg/services/sqlstore/migrator/mysql_dialect.go

```diff
@@ -225,3 +229,76 @@ func (db *MySQLDialect) UpsertSQL(tableName string, keyCols, updateCols []string
 	)
 	return s
 }
+
+func (db *MySQLDialect) Lock() error {
+	sess := db.engine.NewSession()
+	defer sess.Close()
```

Comment:
I'm not very familiar with `xorm` and what their concept of a `Session` is so I'm just doing a sanity check.

If session = connection, then opening and closing a session to obtain a lock does not provide proper locking guarantees, since the lock will be automatically released after `Lock` returns, i.e.:

```go
var db *MySQLDialect

db.Lock() // Lock is created and immediately released
db.Migrate() // Is not protected by the lock
db.Unlock() // no-op
```

Please correct me if I'm wrong.

Reply:
you are totally right! I would swear I had made a change to pass the session and not closing it but I must have missed it in the way.

Reply:
@radiohead I have pushed the changes, if you want to check again!